PCF-ADR-0005-mcad-api-groups-migration.md

```diff
@@ -0,0 +1,113 @@
+# MCAD API Groups Migration Plan
+
+|                |                          |
+| -------------- | ------------------------ |
+| Date           | 07/31/2023               |
+| Scope          |                          |
+| Status         | Proposed                 |
+| Authors        | [Antonin Stefanutti](@astefanutti) |
+| Supersedes     | N/A                      |
+| Superseded by: | N/A                      |
+| Issues         | https://github.com/project-codeflare/multi-cluster-app-dispatcher/issues/352 |
+
+## What
+
+The MCAD APIs are currently defined in the `mcad.ibm.com` and `ibm.com` groups.
+The objective of this document is to propose a plan to migrate them to the `.codeflare.dev` top-level group.
+
+This migration will also be the opportunity to organize the APIs by functional areas, i.e., the `workload.codeflare.dev` group for workload management APIs, and the `quota.codeflare.dev` group for quota management APIs.
+
+## Why
+
+It's conventional for open-source projects to use their domain name as their top-level API group, so DNS is leveraged as a global naming system.
+
+The choice of the current MCAD API groups predates the acquisition of the `codeflare.dev` domain, which explains why they do not follow that convention.
+
+Some of the MCAD APIs, mainly the AppWrapper API, is used by downstream projects, as well as end-users.
+From a client standpoint, the impacts of an API group change imply updating the `apiVersion` field of all the MCAD resources they define, as well as updating the group segment of the REST endpoints, created from the CRDs, that they consume.
+
+A migration plan is needed to provide these clients with a smooth transition to the new API groups.
+
+## Goals
+
+The goal is to inform a _progressive_ migration path, i.e., one that does not require all the downstream projects to handle the impacts in locked steps with MCAD.
+A project like [TorchX](https://github.com/pytorch/torchx), which integrates with MCAD[^1] via the AppWrapper API, has its own release cadence, and it’d be very challenging to have all the components migrated within a single migration cycle, without breaking compatibility.
```

Comment:
I think we should also include KubeRay as downstream component consuming MCAD: https://ray-project.github.io/kuberay/guidance/kuberay-with-MCAD/

Reply:
Done with f7126ca7f30d2e51a090be9b745622a05481836d. It's not a technical impact, nor there are any references to the existing API groups. Thought we can take that opportunity to update the links to the CodeFlare project.

Reply:
If the consensus is to go with the _on-shot_ migration alternative, I'll flip it with the _progressive_ migration proposal.